shared/elements/gaia-header/dist/gaia-header.js

```diff
@@ -1127,8 +1127,9 @@ module.exports = component.register('gaia-header', {
   sumButtonWidths: function(buttons) {
     var defaultWidth = 50;
     var sum = buttons.reduce((prev, button) => {
+      var isEmpty = !button.children.length && !button.textContent;
```

Comment:
I think your code is right here: if it's empty, it likely means we use the default width (we usually use the default width for data-icons). I think Wilson didn't look carefully here :) but empty can mean we left a new line or one space, but still we'd still use the default width in this case.

no, the only downside to your approach is that we don't know if the button is _hidden_. And I don't have a good solution for this...